packages/cells/src/model.ts

```diff
@@ -295,25 +295,55 @@ export class CellModel extends CodeEditor.Model implements ICellModel {
     event: IObservableJSON.IChangedArgs
   ): void {
     const metadata = this.nbcell.getMetadata();
-    this._mutex(() => {
+    this._modeDBMutex(() => {
       switch (event.type) {
         case 'add':
-          metadata[event.key] = event.newValue as any;
+          this._changeCellMetata(metadata, event);
+          break;
+        case 'change':
+          this._changeCellMetata(metadata, event);
           break;
         case 'remove':
           delete metadata[event.key];
           break;
-        case 'change':
-          metadata[event.key] = event.newValue as any;
-          break;
         default:
-          metadata[event.key] = event.newValue as any;
-          break;
+          throw new Error(`Invalid event type: ${event.type}`);
       }
       this.nbcell.setMetadata(metadata);
     });
   }
 
+  private _changeCellMetata(
+    metadata: Partial<nbmodel.ISharedBaseCellMetada>,
+    event: IObservableJSON.IChangedArgs
+  ) {
+    switch (event.key) {
+      case 'jupyter':
+        metadata.jupyter = event.newValue as any;
+        break;
+      case 'collapsed':
+        metadata.collapsed = event.newValue as any;
+        break;
+      case 'name':
+        metadata.name = event.newValue as any;
+        break;
+      case 'scrolled':
+        metadata.scrolled = event.newValue as any;
+        break;
+      case 'tags':
+        metadata.tags = event.newValue as any;
+        break;
+      case 'trusted':
+        metadata.trusted = event.newValue as any;
+        break;
+      case 'slideshow':
```

Comment:
I don't know what `slideshow` is, but if we want to store additional information on cells (e.g ``, `slideshow`, or the `mimeType` that is used by the `codeeditor` package) we don't have to make that part of the `metadata`. If it is not notebook-metadata, we shouldn't mix it in just because the name seems appropriate.

This is something that seriously confused me about the project. `mimetype`, for example, is part of the modeldb metadata but I can't find it in `nbformat`. I think it doesn't even need to be synced between the notebooks because it is provided by the context when a text-file is loaded. 

However, I agree with the above approach to set each value explicitly to avoid leaking of unrelated information (e.g. `mimetype`, or `slideshow` which are not defined in nbformat).

If you want we can meet tomorrow to compare different approaches.

Reply:
mimeType in codeeditor is a very special case that is not to be compared with the rest. I think it is a legacy rest, but let's talk about that tormmorrow.